### PR TITLE
Debug clap cli

### PR DESCRIPTION
**Pull request by zancas**
This code adds parsers at the app/command-line boundary:

* birthday --> u32
* server --> http::Uri
* seed --> string of 24 whitespace delimited elements

It also:
* adds deprecations from zingolib, so that the cli will automatically expose "list"

TODO:

   * Make cli run without cargo.
   * make seed comply with relevant seedphrase-word-library
   * remove crufty post-parse code that's no longer necessary
   * encode commands.rs as clap subcommands

Tests thus far have been manual invocations of:

```
rm -f ~/wallets/sapling-balance-bug/zingo-wallet.dat && cargo run --bin zingo-cli -- --data-dir ~/wallets/sapling-balance-bug/ --seed "daughter safe REDACTED" --birthday 2328948
```

With various values of the birthday and seed values.  I have not yet tested the server argument.

### CODE REVIEW

File: zingocli/src/lib.rs [hunk 1]
```diff
@@ -65,6 +68,23 @@ pub fn build_clap_app() -> clap::ArgMatches {
         ).get_matches()
 }
 
+// Custom function to parse a string into an http::Uri
```
Comment (Oscar-Pepper) [inline zingocli/src/lib.rs:71]: should this be a doc comment?

File: zingocli/src/lib.rs [hunk 2]
```diff
@@ -65,6 +68,23 @@ pub fn build_clap_app() -> clap::ArgMatches {
...
+fn parse_uri(s: &str) -> Result<http::Uri, String> {
+    s.parse::<http::Uri>().map_err(|e| e.to_string())
+}
+// Custom function to parse a string into an http::Uri
```
Comment (Oscar-Pepper) [inline zingocli/src/lib.rs:75]: doc comment?

**Review by Oscar-Pepper — DISMISSED** [review]
can you add to this PR a change to readme with the cargo run command instead of running binary directly?

EDIT: oh i see the TODOs are for this PR sorry NVM!

**Comment by zancas** [comment]
I have demonstrated that the naked binary runs, which I believe addresses the coment.

**Comment by fluidvanadium** [comment]
note from audit, seed should not be input directly from the commandline

**Review by fluidvanadium — APPROVED** [review]
i tested cargo run and ./target/debug/zingo-cli

**Pull request merged**